File: Components/RepositoryComments/RepositoryCommentController.vb
Imports System
Imports System.Configuration
Imports System.Data
Imports System.Data.SqlClient
Imports System.Data.SqlTypes
Imports System.Reflection
Imports System.Xml
Imports DotNetNuke
Imports DotNetNuke.Common.Utilities
Imports DotNetNuke.Common.Globals
Imports DotNetNuke.Services.Search

Namespace DotNetNuke.Modules.Repository
#Region "RepositoryComments"

    Public Class RepositoryCommentController

        Public Function GetRepositoryComments(ByVal ObjectId As Integer, ByVal moduleid As Integer) As ArrayList
            Return CBO.FillCollection(DataProvider.Instance().GetRepositoryComments(ObjectId, moduleid), GetType(RepositoryCommentInfo))
        End Function
        Public Function GetSingleRepositoryComment(ByVal ItemId As Integer, ByVal moduleid As Integer) As RepositoryCommentInfo
            Return CType(CBO.FillObject(DataProvider.Instance().GetSingleRepositoryComment(ItemId, moduleid), GetType(RepositoryCommentInfo)), RepositoryCommentInfo)
        End Function
        Public Function AddRepositoryComment(ByVal ItemId As Integer, ByVal moduleid As Integer, ByVal UserName As String, ByVal Comment As String) As Integer
            Return CType(DataProvider.Instance().AddRepositoryComment(ItemId, moduleid, UserName, Comment), Integer)
        End Function
        Public Sub UpdateRepositoryComment(ByVal ItemId As Integer, ByVal moduleid As Integer, ByVal UserName As String, ByVal Comment As String)
            DataProvider.Instance().UpdateRepositoryComment(ItemId, moduleid, UserName, Comment)
        End Sub
        Public Sub DeleteRepositoryComment(ByVal ItemID As Integer, ByVal moduleid As Integer)
            DataProvider.Instance().DeleteRepositoryComment(ItemID, moduleid)
        End Sub

    End Class

#End Region
End Namespace

File: Components/RepositoryComments/RepositoryCommentController.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Diagnostics;
using System.Drawing;
using System.Linq;
using System.Configuration;
using System.Data.SqlClient;
using System.Data.SqlTypes;
using System.Reflection;
using System.Xml;
using DotNetNuke;
using DotNetNuke.Common.Utilities;
using DotNetNuke.Common;
using DotNetNuke.Services.Search;

namespace DotNetNuke.Modules.Repository
{
	#region "RepositoryComments"

	public class RepositoryCommentController
	{

		public ArrayList GetRepositoryComments(int ObjectId, int moduleid)
		{
			return CBO.FillCollection(DataProvider.Instance().GetRepositoryComments(ObjectId, moduleid), typeof(RepositoryCommentInfo));
		}
		public RepositoryCommentInfo GetSingleRepositoryComment(int ItemId, int moduleid)
		{
            return CBO.FillObject<RepositoryCommentInfo>(DataProvider.Instance().GetSingleRepositoryComment(ItemId, moduleid));
		}
		public int AddRepositoryComment(int ItemId, int moduleid, string UserName, string Comment)
		{
			return Convert.ToInt32(DataProvider.Instance().AddRepositoryComment(ItemId, moduleid, UserName, Comment));
		}
		public void UpdateRepositoryComment(int ItemId, int moduleid, string UserName, string Comment)
		{
			DataProvider.Instance().UpdateRepositoryComment(ItemId, moduleid, UserName, Comment);
		}
		public void DeleteRepositoryComment(int ItemID, int moduleid)
		{
			DataProvider.Instance().DeleteRepositoryComment(ItemID, moduleid);
		}

	}

	#endregion
}
